fw/http_limits.c

```diff
@@ -331,11 +336,25 @@ frang_conn_new(struct sock *sk, struct sk_buff *skb)
 	 * for a single client is absolutely straight-forward.
 	 */
 	r = frang_conn_limit(ra, dflt_vh->frang_gconf);
-	if (unlikely(r == T_BLOCK) && dflt_vh->frang_gconf->ip_block) {
-		tfw_filter_block_ip(cli);
-		tfw_client_put(cli);
-	}
 
+	/*
+	 * In case of closing connection by FIN-ACK, tfw_classify_conn_close()
+	 * is responsible for resources releasing.
+	 * In case of dropping connection by RST, we should release
+	 * resources here.
+	 */
+	if (unlikely(r == T_BLOCK)) {
+		if(dflt_vh->frang_gconf->ip_block) {
```

Comment:
```suggestion
		if (dflt_vh->frang_gconf->ip_block) {
```
Why this code is not in `frang_conn_limit()` to not to acquire the spin lock twice?

Reply:
Short answer: because it was before, I just added else clause.
Long answer: because `frang_conn_limit()` has 4 return points and in this way we catch all possible results.
Sure, not a big problem to rework  `frang_conn_limit()` to make single return point.
Also it located in unlikely clause.
BTW it lead to another very interesting question. `T_BLOCK` considered as less probable limitation result. 
But in case of (D)DoS number of the blocked connections may on several orders of magnitude higher than enabled.
So what clause is really "likely" ?